store/types/gas.go

```diff
@@ -63,21 +63,25 @@ func NewGasMeter(limit Gas) GasMeter {
 	}
 }
 
+// GasConsumed returns the gas consumed from the GasMeter.
 func (g *basicGasMeter) GasConsumed() Gas {
 	return g.consumed
 }
 
+// GasRemaining returns the gas left in the GasMeter.
 func (g *basicGasMeter) GasRemaining() Gas {
 	if g.IsPastLimit() {
 		return 0
 	}
 	return g.limit - g.consumed
 }
 
+// Limit returns the gas limit of the GasMeter.
 func (g *basicGasMeter) Limit() Gas {
 	return g.limit
 }
 
+// GasConsumedToLimit returns the gas limit if gas consumed is past the limit, otherwise it returns the consumed gas.
```

Comment:
Could there be some additional docs for this one? Its a bit odd that gas can be consumed past the limit

Reply:
```suggestion
// GasConsumedToLimit returns the gas limit if gas consumed is past the limit,
// otherwise it returns the consumed gas.
```

Reply:
Why is that odd @ValarDragon? AFAIU, you can consume past the limit if the tx's operations require more gas than what was provided.

Reply:
It seems odd because ConsumeGas panics when it goes past the limit. 

So this limit behavior only gets called when your recovering from a panic, but in that situation you should know from the error type that you went over the limit. So why would this function make things simpler?

Reply:
Good point. We should add that in the comment.